tests/equip.spec.ts

```diff
@@ -0,0 +1,327 @@
+import Catalog_factory from "../types/constructors/catalog_example";
+import Rmrk_factory from "../types/constructors/rmrk_example_equippable_lazy";
+import Rmrk from "../types/contracts/rmrk_example_equippable_lazy";
+import Contract from "../types/contracts/catalog_example";
+import { ApiPromise, WsProvider, Keyring } from "@polkadot/api";
+import { KeyringPair } from "@polkadot/keyring/types";
+import BN from "bn.js";
+import {
+  PartType,
+  Part,
+} from "../types/types-arguments/catalog_example";
+import { RmrkError } from "../types/types-returns/rmrk_example_equippable_lazy";
+import { emit } from "./helper";
+
+const MAX_SUPPLY = 888;
+const BASE_URI = "ipfs://tokenUriPrefix/";
+const COLLECTION_METADATA = "ipfs://collectionMetadata/data.json";
+const CATALOG_METADATA = "ipfs://catalogMetadata/data.json";
+const ONE = new BN(10).pow(new BN(18));
+const PRICE_PER_MINT = ONE;
+const ADMIN_ROLE = 0;
+
+// Create a new instance of contract
+const wsProvider = new WsProvider("ws://127.0.0.1:9944");
+// Create a keyring instance
+const keyring = new Keyring({ type: "sr25519" });
+
+describe("RMRK Equip tests", () => {
+    let catalogFactory: Catalog_factory;
+    let avatarFactory: Rmrk_factory;
+    let swordFactory: Rmrk_factory;
+    let api: ApiPromise;
+    let deployer: KeyringPair;
+    let avatar: Rmrk;
+    let sword: Rmrk;
+    let catalog: Contract;
+    let bob: KeyringPair;
+    let dave: KeyringPair;
+
+    beforeEach(async function(): Promise<void> {
+        api = await ApiPromise.create({ provider: wsProvider, noInitWarn: true});
+        deployer = keyring.addFromUri("//Alice");
+        bob = keyring.addFromUri("//Bob");
+        dave = keyring.addFromUri("//Dave");
+
+        avatarFactory = new Rmrk_factory(api, deployer);
+        avatar = new Rmrk(
+            (
+                await avatarFactory.new(
+                    ["Avatar"],
+                    ["AVA"],
+                    [BASE_URI],
+                    MAX_SUPPLY,
+                    PRICE_PER_MINT,
+                    [COLLECTION_METADATA],
+                    deployer.address,
+                    10
+                )
+            ).address,
+            deployer,
+            api
+        );
+
+        swordFactory = new Rmrk_factory(api, deployer);
+        sword = new Rmrk(
+        (
+            await swordFactory.new(
+                ["Sword"],
+                ["SWRD"],
+                [BASE_URI],
+                MAX_SUPPLY,
+                PRICE_PER_MINT,
+                [COLLECTION_METADATA],
+                dave.address,
+                100
+            )
+        ).address,
+        deployer,
+        api
+        );
+
+        catalogFactory = new Catalog_factory(api, deployer);
+        catalog = new Contract(
+            (
+                await catalogFactory.new([CATALOG_METADATA])
+            ).address,
+            deployer,
+            api
+        );
+    });
+
+    it("Equip/Unequip works", async () => {
```

Comment:
```suggestion
    it("Equip works", async () => {
```
This only tests equip functionality, should I also add tests for unequip in this PR or should that be done separately?